api/nodes/permissions.py

```diff
@@ -74,6 +75,34 @@ def has_object_permission(self, request, view, obj):
             has_auth = parent_node.can_edit(auth)
             return has_auth
 
+class ContributorOrPublicForRelationshipPointers(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, dict)
+        auth = get_user_auth(request)
+        parent_node = obj['self']
+
+        if request.method in permissions.SAFE_METHODS:
+            return parent_node.can_view(auth)
+        elif request.method == 'DELETE':
+            return parent_node.can_edit(auth)
+        else:
+            has_parent_auth = parent_node.can_edit(auth)
+            if not has_parent_auth:
+                return False
+            pointer_nodes = []
+            for pointer in request.data.get('data', []):
+                node = Node.load(pointer['id'])
+                if not node or node.is_folder:
```

Comment:
These 2 cases might need to be handled separately; 404 if the node can't be found and 400 if the node isn't a folder. Yes, @brianjgeiger ?